yt/frontends/gadget/data_structures.py

```diff
@@ -54,39 +52,142 @@ def _byte_swap_32(x):
     return struct.unpack('>I', struct.pack('<I', x))[0]
 
 
-def _compute_header_size(header_spec):
-    return sum(field[1] * np.dtype(field[2]).itemsize for field in header_spec)
+class GadgetBinaryHeader(object):
+    """A convenient interface to Gadget binary header.
+    
+    This is a helper class to facilitate the main dataset and IO classes.
+    The main usage is through the GadgetDataset._header attribute. It is also
+    used stand-alone in GadgetDataset._is_valid method.
+    """
+    _placeholder_keys = ['unused', 'empty']
+
+    def __init__(self, filename, header_spec):
+        self.filename = filename
+        if isinstance(header_spec, string_types):
```

Comment:
You'll need to import `string_types` from `yt.extern.six`.

Reply:
Thanks for the reminder! I see it's already imported in the original code.